src/mbgl/mtl/offscreen_texture.cpp

```diff
@@ -41,8 +41,13 @@ class OffscreenTextureResource final : public RenderableResource {
     }
 
     PremultipliedImage readStillImage() {
-        assert(false);
-        return PremultipliedImage();
+        auto data = std::make_unique<uint8_t[]>(texture->getDataSize());
+        MTL::Region region = MTL::Region::Make2D(0, 0, size.width, size.height);
+        NS::UInteger bytesPerRow = size.width * texture->getPixelStride();
+
+        static_cast<Texture2D*>(texture.get())->getMetalTexture()->getBytes(data.get(), bytesPerRow, region, 0);
```

Comment:
Does this offer implicit synchronization (like GL) or do we need a read barrier?

Reply:
In general Metal doesn't do much of anything implicitly so it probably depends on what we're using this for.
If we're using it to read back an offscreen render buffer, then we want to (and are probably already) waiting for the frame to finish so this would be fine.  In that use case.